# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/BufferReader.h
# Commit: f6df4eb84932
# Full Hash: f6df4eb84932ddaccae4e53d4897234e0153913e
# Author: Chris Pearce <cpearce@mozilla.com>
# Date: 2019-06-14 10:00:22
# Regressor Bug: 1558364
# File Overlap Count: 1
# Description:
#   Bug 1558364 - Add MediaSpan and use it for TrackBuffersManager::mInputBuffer. r=jya
#   
#   As seen in this profile of a Twitch replay: https://perfht.ml/2K9Ydb3 we can
#   often end up spending time in TrackBuffersManager::CodedFrameProcessing()
#   shaving off bytes from the front off TrackBuffersManager::mInputBuffer. This
# ==============================================================================

diff -r e708fe457a95 -r f6df4eb84932 dom/media/BufferReader.h
--- a/dom/media/BufferReader.h	Fri Jun 14 00:37:25 2019 +0000
+++ b/dom/media/BufferReader.h	Fri Jun 14 02:10:09 2019 +0000
@@ -9,6 +9,7 @@
 #include "nscore.h"
 #include "nsTArray.h"
 #include "MediaData.h"
+#include "MediaSpan.h"
 #include "mozilla/Logging.h"
 #include "mozilla/Result.h"
 
@@ -34,6 +35,10 @@
       : mPtr(aData->Elements()),
         mRemaining(aData->Length()),
         mLength(aData->Length()) {}
+  explicit BufferReader(const mozilla::MediaSpan& aData)
+      : mPtr(aData.Elements()),
+        mRemaining(aData.Length()),
+        mLength(aData.Length()) {}
 
   void SetData(const nsTArray<uint8_t>& aData) {
     MOZ_ASSERT(!mPtr && !mRemaining);